extensions/mssql/src/util/dataCache.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export class DataCache<T> {
+
+	millisecondsToLive: number;
+	getValueFunction: (...args: any[]) => T;
+	cache: T;
+	fetchDate: Date;
+
+	constructor(getValueFunction: (...args: any[]) => T, secondsToLive: number) {
+		this.millisecondsToLive = secondsToLive * 1000;
+		this.getValueFunction = getValueFunction;
+		this.cache = undefined;
+		this.fetchDate = new Date(0);
+	}
+
+	public isCacheExpired(): boolean {
+		return (this.fetchDate.getTime() + this.millisecondsToLive) < new Date().getTime();
+	}
+
+	public getData(...args: any[]): T {
+		if (!this.cache || this.isCacheExpired()) {
```

Comment:
Seems like you could just initialize cache to [] and get rid of this check - the cache is going to be expired the first call anyways.

Reply:
I am not quite sure I understand. The cached item is of type T so I cannot initialize it to an empty array, right? I would see something like `Type 'undefined[]' is not assignable to type 'T'.`

Reply:
You refactored it so this isn't relevant anymore, but initializing it doesn't require the type. You would have just made the definition like this :

`private cache: T[] = [];`